packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -112,14 +115,53 @@ const protoConverterCache = createCache(
 
 const isAfterFeb2024OrNewApis = false;
 
-/**
- * @param objs the objects to be converted, the contents of this array will be mutated
+/*
+ * Currently loses all information about the interface. Future PRs will improve
+ * FIXME
  */
-export async function convertWireToOsdkObjectsInPlace(
+export async function convertWireToOsdkInterfaceInPlace(
   client: MinimalClient,
-  objs: OntologyObjectV2[],
+  objects: OntologyObjectV2[],
+  interfaceApiName: string,
 ) {
-  // Fix properties
+  fixObjectPropertiesInline(objects);
+
+  const localObjectCache = createAsyncCache((client, apiName: string) =>
+    client.ontology.provider.getObjectDefinition(apiName)
+  );
+
+  const uniqueApiNames = new Set<string>(objects.map(o => o.$apiName));
+  const [interfaceType] = await Promise.all([
+    client.ontology.provider.getInterfaceDefinition(interfaceApiName),
+    ...Array.from(uniqueApiNames).map(n => localObjectCache.get(client, n)),
```

Comment:
Do you want to do `Array.from(uniqueApiNames).map(n=>localObjectsCache.get(client, n))` above this await without blocking on it? That would let us proceed with conversions until we run into one of these that hasn't loaded.

Its probably not worth bucketing on apiName because you'd lose ordering.

Reply:
Improved per suggestion